swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -651,6 +664,136 @@ private module OutNodes {
 
 import OutNodes
 
+private predicate closureFlowStep(Expr e1, Expr e2) {
+  // simpleLocalFlowStep(exprNode(e1), exprNode(e2)) // TODO: find out why the java version uses simpleAstFlowStep... probably due to non-monotonic recursion
```

Comment:
That's not quite the reason. We want a simple value flow relation that enables us to find the "interesting" aliases of a closure. We primarily want to use SSA steps to identify those aliases that arise from assignment to a variable. And we want to include steps like casts (those are the AST steps). For identifying the "interesting" flow points, we want to find those points that "escape", i.e. argument positions, return positions, and the like. If we use def-use rather than use-use, then we have a very simple way to achieve this, namely by returning those aliases that don't step any further.  That allows the very simple implementation of `hasAliasedAccess(Expr f)` instead of having to somehow enumerate possibly escape points.

To understand why we want this from the `hasAliasedAccess` relation, we need to look at what the shared lib uses this for: There are two purposes. 1. update the value of the captured variables just-in-time before a closure escapes to ensure that it has the latest value. 2. observe side-effects when a closure is called - this requires that all the aliases that are used either directly as a call or occurs in an argument position of a call that may perform a callback are included in the `hasAliasedAccess` relation.